src/Microsoft.SqlTools.Migration/MigrationService.cs

```diff
@@ -580,6 +583,62 @@ public void InitializeService(IProtocolEndpoint serviceHost)
             }
         }
 
+        internal async Task HandleGetArmTemplateRequest(
+    string targetType,
+    RequestContext<string> requestContext)
+        {
+            ProvisioningScriptServiceProvider provider = new ProvisioningScriptServiceProvider();
+
+            List<SkuRecommendationResult> recommendations = ExtractSkuRecommendationReportAction.ExtractSkuRecommendationsFromReport(GetReportFilePath(SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, targetType));
+
+            SqlArmTemplate template = provider.GenerateProvisioningScript(recommendations);
+
+            string jsonOutput = JsonConvert.SerializeObject(
+                template,
+                Formatting.Indented,
+                new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore }
+            );
+
+            await requestContext.SendResult(jsonOutput);
+
+        }
+
+        /// <summary>
+        /// Scans the specified output folder for a SkuRecommendationReport*.json file, used when no report file path is specified 
+        /// </summary>
+        /// <param name="outputFolder"></param>
+        /// <returns>The file path to the SKU recommendation report, if it exists</returns>
+        public static string GetReportFilePath(string outputFolder, string targetType)
+        {
+            string skuRecommendationJsonRegex = "SkuRecommendationReport-";
+            switch (targetType)
+            {
+                case "AzureSqlDatabase":
+                    skuRecommendationJsonRegex += "AzureSqlDatabase-Baseline*.json";
+                    break;
+                case "AzureSqlManagedInstance":
+                    skuRecommendationJsonRegex += "AzureSqlManagedInstance-Baseline*.json";
+                    break;
+                case "AzureSqlVirtualMachine":
+                    skuRecommendationJsonRegex += "AzureSqlVirtualMachine-Baseline*.json";
+                    break;
+
+            }
+
+            string filePath = Directory.GetFiles(outputFolder, skuRecommendationJsonRegex).FirstOrDefault();
```

Comment:
it seams this call should handle to pick the latest one. Just test it.

Reply:
this was not picking latest, but fixed it.